Gs sec 2 fp review

gselander opened the pull request:
Proposed resolution to Francesca's review comments on Section 2.

gselander committed:
rephrased requirements for verifying Echo option value

gselander committed:
text on event-based freshness

gselander committed:
clarifying same Echo value in different responses

gselander committed:
replaced 'fresh enough' with 'fresh'

gselander committed:
clarify Echo option value must be i.p.

gselander committed:
clarifying 'property'

chrysn commented on draft-ietf-core-echo-request-tag.md:
```diff
@@ -138,15 +138,15 @@ The Echo option MAY be included in any request or response (see {{echo-app}} for
+If a certain request is required to be fresh, the request does not contain a fresh Echo option value, and the server cannot verify the freshness of the request in some other way, the server MUST NOT process the request further and SHOULD send a 4.01 Unauthorized response with an Echo option. The server MAY include the same Echo option value in several different response messages and to different  ...
 
-The server may use request freshness provided by the Echo option to verify the aliveness of a client or to synchronize state. The server may also include the Echo option in a response to force a client to demonstrate reachability at its claimed network address.
+The server may use request freshness provided by the Echo option to verify the aliveness of a client or to synchronize state. The server may also include the Echo option in a response to force a client to demonstrate reachability at its claimed network address. Note that the Echo option does not bind a request uniquely to a previous response, but provides an indication that the client had access  ...
```
I like that addition a lot.

emanjon replied:
I would suggest removing "uniquely".

chrysn replied:
Maybe "to a particular previous"?

chrysn commented on draft-ietf-core-echo-request-tag.md:
```diff
@@ -174,7 +174,15 @@ Client   Server
+* All cache values MUST be different, or different with a probability close to 1 so they are different for all practical purposes.
+* The oldest value in the cache is overwritten at new events.
+
+The message flow in Figure 2 can serve as illustration also of event-based freshness. In this case t0 is defined as the most recent value in the cache when the server sends the 4.01 (Unauthorized) response. The Echo option value is echoed by the client in the second request and compared by the server with the contents of the cache at the reception of the second request. If t0 is in the cache then ...
```
Given we're up to reorder text it'd be useful to have kramdown resolve that number and refer to it by anchor in the source.

chrysn commented on draft-ietf-core-echo-request-tag.md:
```diff
@@ -174,7 +174,15 @@ Client   Server
 {: #echo-figure title="Example Echo Option Message Flow" artwork-align="center"}
 
-When used to serve freshness requirements (including client aliveness and state synchronizing), CoAP messages containing the Echo option MUST be integrity protected between the intended endpoints, e.g. using DTLS, TLS, or an OSCORE Inner option ({{RFC8613}}). When used to demonstrate reachability at a claimed network address, the Echo option SHOULD contain the client's network address, but MAY be ...
+Another way for the server to verify freshness is to maintain a cache of values associated to events: 
```
I understand this to be a usage example. At this point I think it'd be easier for the reader to digest if this were simplified to a single latest-state (cache size 1), and explained in terms of "fresh" and "not fresh" rather than taking up the t0/t1 terminology of the time-based example.

emanjon replied:
I like the addition but I think there are to much implementation details for this section. That kind of stuff we have in the appendix. I also think all time should be removed from the event based example.

One solution could be to udate the figure to

OLD     Echo: 0x437468756c687521 (t0)
NEW     Echo: 0x437468756c687521 (t0 or event value)

Alt.

OLD     Echo: 0x437468756c687521 (t0)
NEW     Echo: 0x437468756c687521 (t0 or e0)

chrysn reviewed (approved):
Happy with the changes even if you turn down the two remarks I made.

emanjon commented on draft-ietf-core-echo-request-tag.md:
```diff
@@ -138,15 +138,15 @@ The Echo option MAY be included in any request or response (see {{echo-app}} for
 The application decides under what conditions a CoAP request to a resource is required to be fresh. These conditions can for example include what resource is requested, the request method and other data in the request, and conditions in the environment such as the state of the server or the time of the day.
 
-If a certain request is required to be fresh, the request does not contain a fresh Echo option value, and the server cannot verify the freshness of the request in some other way, the server MUST NOT process the request further and SHOULD send a 4.01 Unauthorized response with an Echo option. The server MAY include the same Echo option value in several different responses and to different clients.
+If a certain request is required to be fresh, the request does not contain a fresh Echo option value, and the server cannot verify the freshness of the request in some other way, the server MUST NOT process the request further and SHOULD send a 4.01 Unauthorized response with an Echo option. The server MAY include the same Echo option value in several different response messages and to different  ...
```
I think it is more important to have an example of time based to not give the idea that only event based echo values can be reused. The server is perfectly fine to send the same echo value to to clients whos requests came close in time. This saves some processing.

emanjon commented on draft-ietf-core-echo-request-tag.md:
```diff
@@ -189,7 +197,7 @@ The CoAP server side of CoAP-to-HTTP proxies MAY request freshness, especially i
     * For actuator applications with low delay tolerance, to avoid additional round-trips for multiple requests in rapid sequence, the server may include the Echo option with a new value even in a successful response to a request, irrespectively of whether the request contained an Echo option or not. The client then uses the Echo option with the new value in the next actuation request, and the se ...
 
-2. A server may use the Echo option to synchronize state or time with a requesting client. A server MUST NOT synchronize state or time with clients which are not the authority of the property being synchronized. E.g. if access to a server resource is dependent on time, then the client MUST NOT set the time of the server.
+2. A server may use the Echo option to synchronize a property (state or time) with a requesting client. A server MUST NOT synchronize a property with clients which are not the authority of the property being synchronized. E.g. if access to a server resource is dependent on time, then the client requesting access MUST NOT set the time of the server.
```
OLD a property (state or time) 
NEW properties (such as state or time)

emanjon committed:
Update draft-ietf-core-echo-request-tag.md

Removed unique

Can sync several properties with a single echo. Not sure that state and time is the complete space.

emanjon committed:
Update draft-ietf-core-echo-request-tag.md

Added time based example as well
Removed "see later in this section" as the terms are defined earlier in the document.

emanjon commented on draft-ietf-core-echo-request-tag.md:
```diff
@@ -189,7 +197,7 @@ The CoAP server side of CoAP-to-HTTP proxies MAY request freshness, especially i
 
     * For actuator applications with low delay tolerance, to avoid additional round-trips for multiple requests in rapid sequence, the server may include the Echo option with a new value even in a successful response to a request, irrespectively of whether the request contained an Echo option or not. The client then uses the Echo option with the new value in the next actuation request, and the se ...
 
```
"E.g. if access to a server resource is dependent on time, then the client requesting access MUST NOT set the time of the server."

Two comments of this sentence.
1. This seems to rely on that the server is not the authority (which is may be)
1. A client syncronizing seems to have nothing to do with echo

Commit:
wording: binding to "a unique" -> "any particular" request

See-Also: https://github.com/core-wg/echo-request-tag/pull/49#discussion_r341536810

gselander committed:
resolving comments on PR49

gselander committed:
resolving comment on property

emanjon committed:
changes hex string so that e0 and t0 are different
